### PR TITLE
feat(GlobalHeaderItems): Business Sso check for profile

### PR DESCRIPTION
**Pull request by jenesh**
### Overview

Adding a check to hide the account + billing for all Business SSO users including all Skillsoft users. This is because users that are SSO'd don't need the other account options besides Goal settings.

PR related to this change: https://github.com/codecademy-engineering/Codecademy/pull/30354

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: [EN-1689](https://codecademy.atlassian.net/browse/EN-1689)
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.99-alpha.d775dd.0<br/>@codecademy/gamut-labs@31.3.4-alpha.d775dd.0<br/>@codecademy/styleguide@56.2.4-alpha.d775dd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6255bd2518c8e600b90f2b22--gamut-preview.netlify.app>https://6255bd2518c8e600b90f2b22--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6255bd2518c8e600b90f2b22>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.99-alpha.ffbe19.0<br/>@codecademy/gamut-labs@31.3.4-alpha.ffbe19.0<br/>@codecademy/styleguide@56.2.4-alpha.ffbe19.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6255eb724d91ed00849b925e--gamut-preview.netlify.app>https://6255eb724d91ed00849b925e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6255eb724d91ed00849b925e>Deploy Logs</a></p>

**Review by CharlieBliss — APPROVED** [review]

**Review by dreamwasp — APPROVED** [review]

File: packages/gamut-labs/src/GlobalHeader/GlobalHeaderItems.tsx [hunk 1]
```diff
@@ -278,7 +285,14 @@ export const proProfile = (
   user: User,
   isMobile?: boolean
 ): AppHeaderProfileDropdownItem => {
-  const topSection = [profileMyProfile, profileAccount, profileMyHome];
+  const topSection = [profileMyProfile];
+
+  if (!user.isBusinessSsoUser) {
+    topSection.push(profileAccount);
```
Comment (antoniablair) [inline packages/gamut-labs/src/GlobalHeader/GlobalHeaderItems.tsx:291]: not blocking, but I could see us wanting to keep this for staff `business_admin` users

Comment (jenesh) [reply packages/gamut-labs/src/GlobalHeader/GlobalHeaderItems.tsx:291]: The `isBusinessSsoUser` is only true if that user has been SSO and so far there are no `business_admin`'s that have been SSO'd to my knowledge but I'll update so all business admins still have this option.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.99-alpha.65cc0e.0<br/>@codecademy/gamut-labs@31.3.4-alpha.65cc0e.0<br/>@codecademy/styleguide@56.2.4-alpha.65cc0e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://625821d0814aa01b19acdb57--gamut-preview.netlify.app>https://625821d0814aa01b19acdb57--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/625821d0814aa01b19acdb57>Deploy Logs</a></p>

**Pull request merged**